qutip/tests/test_random.py

```diff
@@ -91,5 +91,26 @@ def test_rand_super_dims():
         check_func_dims(func, (7, ), {}, [[[7], [7]]] * 2)
         check_func_dims(func, (6, ), {'dims': [[[2, 3], [2, 3]], [[2, 3], [2, 3]]]}, [[[2, 3], [2, 3]], [[2, 3], [2, 3]]])
 
+def test_rand_ket():
+    """
+    Random Qobjs: Tests that N and dims are determined as needed.
+    """
+    # when neither N or dims are specified
+    assert_raises(ValueError,rand_ket, )
+
+    # when both N and dims are specified
+    test_both_N_dim = rand_ket(N=7,dims=[[7], [1]])
+    assert(np.shape(test_both_N_dim)==(7,1))
+
+    # when dims is specified and N has to be determined
+    test_only_dim = rand_ket(dims=[[7], [1]])
+    assert(np.shape(test_only_dim)==(7,1))
+
+    # when N is specified but dims has to be defined using default
+    test_only_N = rand_ket(N=7)
+    assert(np.shape(test_only_N)==(7,1))
```

Comment:
I suggest you actually merge these cases into the test above, but a couple of additional points too:
1. can you add some cases that test `N` is calculated correctly including implicit tensor structure in `dims`, because that's what this is meant to be most useful for, right?
2. `np.shape` is for NumPy arrays, not `Qobj` (the fact that it works is kind of unintended side-effect).  `Qobj` already provides `Qobj.shape` to access this information directly.  
3. a very minor quibble, but `assert` isn't a function, so it doesn't need the brackets